src/main/java/com/hongliangjie/fugue/topicmodeling/LDA/LDA.java

```diff
@@ -210,38 +211,155 @@ protected void initTrainModel() {
         LOGGER.info("Finished initializing model");
     }
 
-    public double likelihood(int modelID) {
+    protected abstract class HyperparameterOptimization{
+        public abstract void Optimize(); // as hyper-parameter optimization only happens in training, the modelID is always set to 0 and therefore ignored.
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Method names should not start with capital letters](https://www.codacy.com/app/hongliangjie/Fugue/file/2346060949/issues/source?bid=3162994&fileBranchId=3208469#l215)